### PR TITLE
tracer: Improve test coverage for x-ray

### PR DESCRIPTION
**Pull request by marcomagdy**
- Removed unused functions that were showing up untested
- Renamed a function to match the style guide
- Small refactor (now that I understand how to use buffers)
- Added a test that exercises the sending bytes to the x-ray daemon

Risk Level: low
Testing: unit tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: source/extensions/tracers/xray/daemon_broker.cc [hunk 1]
```diff
@@ -40,11 +36,9 @@ void DaemonBrokerImpl::send(const std::string& data) const {
   constexpr auto version = 1;
   constexpr auto format = "json";
   const std::string payload = absl::StrCat(createHeader(format, version), "\n", data);
-  Buffer::RawSlice buf;
-  buf.mem_ = const_cast<char*>(payload.data());
-  buf.len_ = payload.length();
-  const auto rc = Network::Utility::writeToSocket(*io_handle_, &buf, 1 /*num_slices*/,
-                                                  nullptr /*local_ip*/, *address_);
+  Buffer::OwnedImpl buf(payload);
```
Comment (mattklein123) [inline source/extensions/tracers/xray/daemon_broker.cc:39]: Note this will copy which might not be what you want.

Comment (marcomagdy) [reply source/extensions/tracers/xray/daemon_broker.cc:39]: ugh - yeah. I'll revert that.

File: test/extensions/tracers/xray/tracer_test.cc [hunk 2]
```diff
@@ -183,6 +183,16 @@ TEST_F(XRayTracerTest, TraceIDFormatTest) {
   ASSERT_EQ(24, parts[2].length());
 }
 
+TEST(XRayDaemon, SendBytesOverUDP) {
```
Comment (mattklein123) [inline test/extensions/tracers/xray/tracer_test.cc:186]: This doesn't actually verify anything, right? Can you look at the tests I added for statsd not that long ago? You should be able to get a real working UDP test running quite quickly which will give you real coverage.

Comment (marcomagdy) [reply test/extensions/tracers/xray/tracer_test.cc:186]: It doesn't verify anything. I was mainly trying to get it to run through the sanitizers to flush out any memory safety bugs.

I'll look at the statsd tests

Comment (marcomagdy) [reply test/extensions/tracers/xray/tracer_test.cc]: I added a UDP test. I had to make a small change to the utility class to make it listen on ports other than loopback.

File: test/test_common/network_utility.h [hunk 3]
```diff
@@ -180,8 +180,11 @@ Api::IoCallUint64Result readFromSocket(IoHandle& handle, const Address::Instance
  */
 class UdpSyncPeer {
 public:
+  // Create a peer that listens on the given version of the loopback address.
   UdpSyncPeer(Network::Address::IpVersion version);
 
+  // Create a peer that listens on the given address:port.
+  UdpSyncPeer(Network::Address::InstanceConstSharedPtr address);
```
Comment (mattklein123) [inline test/test_common/network_utility.h:187]: For tests we can't bind to a particular port. Similar to the UDP statsd tests, you should create a target which will bind to port 0 and then determine that actual port, and then feed that into your test.

File: test/extensions/tracers/xray/tracer_test.cc [hunk 4]
```diff
@@ -183,6 +184,41 @@ TEST_F(XRayTracerTest, TraceIDFormatTest) {
   ASSERT_EQ(24, parts[2].length());
 }
 
+TEST(XRayDaemon, VerifyUdpPacketContents) {
+  NiceMock<Server::MockInstance> server;
+  Network::Test::UdpSyncPeer xray_fake_daemon(Network::Address::IpVersion::v4);
```
Comment (mattklein123) [inline test/extensions/tracers/xray/tracer_test.cc:189]: Apologies but this needs to be parameterized on the supported IP types, as some people don't support v4 or run tests there. Please see the UDP example again for how to do this.

Comment (marcomagdy) [reply test/extensions/tracers/xray/tracer_test.cc]: No problem. I'm learning a lot this way.

**Pull request merged**